sqlmesh/dbt/builtin.py

```diff
@@ -162,13 +171,33 @@ def var(name: str, default: t.Optional[t.Any] = None) -> t.Any:
     return var
 
 
-def generate_ref(refs: t.Dict[str, t.Any], api: Api) -> t.Callable:
+def generate_ref(
+    refs: t.Dict[str, t.Any],
+    api: Api,
+    snapshots: t.Dict[str, Snapshot],
+    is_dev: bool,
+    dialect: t.Optional[str],
+) -> t.Callable:
     def ref(package: str, name: t.Optional[str] = None) -> t.Optional[BaseRelation]:
         ref_name = f"{package}.{name}" if name else package
         relation_info = refs.get(ref_name)
         if relation_info is None:
+            logger.warning("Could not resolve ref '%s'", ref_name)
             return None
 
+        relation_info = relation_info.copy()
+        model_name = ".".join(
+            relation_info[p] for p in ("database", "schema", "identifier") if relation_info.get(p)
+        )
+        table_mapping = to_table_mapping(snapshots.values(), is_dev)
+        if model_name in table_mapping:
+            physical_table_name = table_mapping[model_name]
```

Comment:
Is it possible that the mapping isn't the dict? If so, should we catch and raise a better error message?

Reply:
When mapping can not be a dict?

Reply:
I meant the mapping isn't *in* the dict. Key word missing there!